components/SearchResult.tsx

```diff
@@ -0,0 +1,28 @@
+import discogs from "../app/utils/discogs";
+import React from "react";
+import { useQuery } from "@tanstack/react-query";
+
+export default function SearchResult({ search }: { search: string }) {
+  const { data, isError, isLoading } = useQuery({
+    queryKey: ["search"],
+    queryFn: async () => {
+      const params = new URLSearchParams({ search: "the strokes" });
+      return fetch("/api/search?" + params).then((res) => res.json());
```

Comment:
if you look at the result type though, it's `any`:

![CleanShot 2023-09-01 at 14 38 15](https://github.com/Rmaki27/collections/assets/37197876/6d10ac11-9a05-4c04-ba27-b662543ff2f4)

in order to narrow the type, we need to add actual runtime JS code to ensure it is what we expect.

Luckily that's why we wrote the zod schema for the search response, so we can just import that and use it here:

```ts
const parsedResult = SearchResponseSchema.parse(result)
```

if you look at the type of `parsedResult`, it's actually what we want.

That's because the `parse` function will validate the `result` object at runtime, and throw an error if they don't match (therefore it's impossible `parsedResult` could be anything other than `SearchResponseSchema`, and thanks to Zod, Typescript knows that).

React query also types `data` as the return type of this function, so the type of `data` should be correct now too.